static/js/message_flags.js

```diff
@@ -113,6 +113,18 @@ exports.toggle_starred_and_update_server = function (message) {
     }
 };
 
+exports.unstar_all_messages = function () {
+    var starred_msg_ids = page_params.starred_messages;
+    _.each(starred_msg_ids, function (msg_id) {
+        var msg = message_store.get(msg_id);
+        msg.starred = false;
+        unread_ops.notify_server_message_read(msg);
+        ui.update_starred_view(msg.id, false);
+        send_flag_update(msg, 'starred', 'remove');
```

Comment:
We should replace this with a bulk call, rather than a loop.  The backend view function already supports a bulk operation ...  we just need to change the JS to construct an appropriate payload.  I think probably making `send_flag_update` take a list of messages is a reasoanble refactoring to support this (and just make all of its existing callers pass a 1-element list, e.g. `[message]` .